plugins/languages/treesitter/treesitter.nix

```diff
@@ -6,210 +6,169 @@
   ...
 }:
 with lib;
-let
-  cfg = config.plugins.treesitter;
-in
-{
-  options = {
-    plugins.treesitter = {
-      enable = mkEnableOption "tree-sitter syntax highlighting";
-
-      package = mkOption {
-        type = types.package;
-        default = pkgs.vimPlugins.nvim-treesitter;
-        description = "Plugin to use for nvim-treesitter. If using nixGrammars, it should include a `withPlugins` function";
-      };
-
-      nixGrammars = mkOption {
-        type = types.bool;
-        default = true;
-        description = "Install grammars with Nix";
-      };
-
-      ensureInstalled = mkOption {
-        type =
-          with types;
-          oneOf [
-            (enum [ "all" ])
-            (listOf str)
-          ];
-        default = "all";
-        description = "Either \"all\" or a list of languages";
-      };
-
-      gccPackage = helpers.mkPackageOption {
-        default = if cfg.nixGrammars then null else pkgs.gcc;
-        description = ''
-          Which package (if any) to be added as the GCC compiler.
-          This is required to build grammars if you are not using `nixGrammars`.
-          To disable the installation of GCC, set this option to `null`.
-        '';
-      };
-
-      parserInstallDir = mkOption {
-        type = types.nullOr types.str;
-        default = if cfg.nixGrammars then null else "$XDG_DATA_HOME/nvim/treesitter";
-        description = ''
-          Location of the parsers to be installed by the plugin (only needed when nixGrammars is disabled).
-          This default might not work on your own install, please make sure that $XDG_DATA_HOME is set if you want to use the default. Otherwise, change it to something that will work for you!
-        '';
-      };
-
-      ignoreInstall = mkOption {
-        type = types.listOf types.str;
-        default = [ ];
-        description = "List of parsers to ignore installing (for \"all\")";
-      };
-
-      disabledLanguages = mkOption {
-        type = types.listOf types.str;
-        default = [ ];
-        description = "A list of languages to disable";
-      };
-
-      customCaptures = mkOption {
-        type = types.attrsOf types.str;
-        default = { };
-        description = "Custom capture group highlighting";
-      };
-
-      incrementalSelection =
+helpers.neovim-plugins.mkNeovimPlugin config {
+  name = "treesitter";
+  originalName = "nvim-treesitter";
+  defaultPackage = pkgs.vimPlugins.nvim-treesitter;
+
+  optionsRenamedToSettings = [
+
+  ];
```

Comment:
You can probably copy-paste the _defaults_ test into here, then run a vim macro to format as follows:

```nix
[
  "someOption"
  [ "some" "nested" "option" ]
  "anotherTopLevelSetting"
]
```

Note, use the "old" camelCase names; mkNeovimPlugin will convert the names to snake\_case automatically.

Check the renames work with the tests _before_ updating the tests to the new option names.

Reply:
Good suggestion, I was wondering how to best tackle that, too.

Reply:
Don't be afraid to use `mkRemovedOptionModule` or `mkRenamedOptionModule` for anything that `optionsRenamedToSettings` can't handle.